app/factory.js

```diff
@@ -0,0 +1,37 @@
+const express = require('express');
+const join = require('path').join;
+
+const config = require('./config');
+
+
+module.exports = function (config_override) {
+
+  const app = express();
+
+  let conf = Object.assign(config, config_override);
+
+  init_app(app, conf);
+  init_middleware(app, conf);
+
+  return app;
+};
+
+
+function init_app(app, conf) {
+  app.set('env', conf.app.env);
+
+  const nunjucks = require('nunjucks');
+  nunjucks.configure(join(__dirname, 'templates'), {
+    autoescape: true,
+    express: app
+  });
+}
+
+
+function init_middleware(app, conf) {
+  const middleware = require('./middleware');
```

Comment:
I would rename this module `middleware` to `middlewares` as this is the "map" with the middleware**s**, I'd also rename `contig.middleware` to the plural.

You may want to rename other variables to make the distinction between strings and functions a bit more explicit (I didn't realise about this distinction when I started to have a look at this code):

```javascript
const middlewareNames = require('./middleware');
middlewareNames.forEach(name => {
  const middleware = config.middlewares[name];
  app.use(middleware(app, config));
});
```

Reply:
The word `middleware` is already plural.

In your snippet here, you're confusing the actual middleware (the objects in the middleware module) with the config array, so I take your point.

I'll try to name these better.